Startup.cs

```diff
@@ -0,0 +1,33 @@
+﻿using DI_Example.Services;
+
+namespace DI_Example
+{
+    public class Startup
+    {
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddScoped<ICustomerService, CustomerService>();
```

Comment:
why did you use scoped to register the service?

Reply:
Transient objects are always different; a new instance is provided to every controller and every service.

Scoped objects are the same within a request, but different across different requests.

Singleton objects are the same for every object and every request.

I mean, no difference if i use Scoped or Singleton or Transient in this example